LiveSplit.TR3.asl

```diff
@@ -144,19 +144,32 @@ reset
 
 split
 {
-    // Determine if a split should occur.
     bool shouldSplit;
-    if (current.level == 19)
+    if (current.level == 19 && old.level == 19)
+    {
+        // While playing the "last" level (regardless of IL/FG and if the bonus level, All Hallows, will be played)
+        // the timer should split when the player's control ends and the helicopter scene begins.
+        // This occurs when the player hits the end level trigger which sets levelComplete to true.
         shouldSplit = current.levelComplete && !old.levelComplete;
+    }
     else if (settings["IL"])
+    {
+        // For IL timing, the timer should stop when the end-level stats screen appears.
+        // Checking levelComplete ensures this check works in all cases.
         shouldSplit = current.levelComplete && current.isStatsScreen && !old.isStatsScreen;
+    }
     else
+    {
+        // An easy way to check if the player has advanced into the next level is by checking currentLevelTime == 0.
+        // Because it is impossible to pause and make a save within the first few frames of a level, this check
+        // will not fail if the player happens to loads a save into the next level.
```

Comment:
This else block = Not the last level and not IL: split when timer is reset to 0.

I'd delete the comment.

Reply:
I cut this explanation down to two lines; I believe the explanation about saves and timer could be beneficial. I'm not opposed to a complete removal, though.